### PR TITLE
EPMRPP-33451 React. Dashboard. Widget. Launch statistics

### PR DESCRIPTION
**Pull request by AmsterGet**

### CODE REVIEW

File: app/src/components/widgets/charts/launchStatisticsChart/launchStatisticsChart.jsx [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+    },
+    type: TEST_ITEM_PAGE,
+  });
+
+  getCoords = ({ pageX, pageY }) => {
+    this.x = pageX;
+    this.y = pageY;
+  };
+
+  getConfig = () => {
```
Comment (DiscoElevator) [inline app/src/components/widgets/charts/launchStatisticsChart/launchStatisticsChart.jsx]: Is it possible to split this function into a few smaller ones? It's just difficult to understand what happens here.

File: app/src/components/widgets/charts/common/messages.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    id: 'Charts.failed',
+    defaultMessage: 'Failed',
+  },
+  skipped: {
+    id: 'Charts.skipped',
+    defaultMessage: 'Skipped',
+  },
+  cases: {
+    id: 'Charts.cases',
+    defaultMessage: 'cases', // сценарии
```
[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/components/widgets/charts/common/messages.js]: Fixed

File: app/src/components/widgets/charts/common/tooltip/tooltipContent/tooltipContent.jsx [hunk 3]
```diff
@@ -15,13 +15,16 @@ export const TooltipContent = ({
...
+      {launchName}
+      {launchNumber && ` #${launchNumber}`}
+    </div>
+    {startTime && <div className={cx('launch-start-time')}>{dateFormat(startTime)}</div>}
     <div className={cx('item-wrapper')}>
       <span className={cx('color-mark')} style={{ backgroundColor: color }} />
       <span className={cx('item-name')}>{`${itemName}:`}</span>
       <span className={cx('item-cases')}>
-        <span>{`${itemCases}%`}</span>
+        <span>{itemCases}</span>
```
[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/components/widgets/charts/common/tooltip/tooltipContent/tooltipContent.jsx]: Was fixed if it was affected. (in `app/src/components/widgets/charts/launchesComparisonChart`)

File: app/src/components/widgets/charts/launchStatisticsChart/launchStatisticsChart.jsx [hunk 4]
```diff
@@ -0,0 +1,427 @@
...
+  getDefaultLinkParams = (testItemIds) => ({
+    payload: {
+      projectId: this.props.project,
+      filterId: 'all',
+      testItemIds,
+    },
+    type: TEST_ITEM_PAGE,
+  });
+
+  getCoords = ({ pageX, pageY }) => {
```
[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/components/widgets/charts/launchStatisticsChart/launchStatisticsChart.jsx]: Ok. I will change it also in all charts

File: app/src/components/widgets/charts/launchStatisticsChart/launchStatisticsChart.jsx (hunk 1 above)
Comment (AmsterGet) [reply app/src/components/widgets/charts/launchStatisticsChart/launchStatisticsChart.jsx]: Done

File: app/src/pages/inside/dashboardItemPage/widgetsGrid/widget/widget.jsx [hunk 5]
```diff
@@ -57,6 +64,7 @@ export class Widget extends Component {
 
     this.state = {
       loading: true,
+      show: true,
```
[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/pages/inside/dashboardItemPage/widgetsGrid/widget/widget.jsx]: Fixed

File: app/src/components/widgets/charts/launchStatisticsChart/launchStatisticsChart.scss [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+            }
+          }
+        }
+      }
+    }
+  }
+}
+
+:global {
+  // TODO: fix fulscreen mode for this chart
```
[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/components/widgets/charts/launchStatisticsChart/launchStatisticsChart.scss]: Fixed

File: app/src/components/widgets/charts/common/utils.js [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+  }
+  return (
+    (defectTypeConfig.find((item) => item.locator === locator) || {}).longName ||
+    formatMessage(messages.total, {
+      type: formatMessage(messages[defectTypeConfig[0].shortName.toLowerCase()]),
+    })
+  );
+};
+
+export const getLaunchAxisTicks = (itemsLength) =>
```
[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/components/widgets/charts/common/utils.js]: Done

File: app/src/components/widgets/charts/common/utils.js [hunk 8]
```diff
@@ -0,0 +1,76 @@
+import * as COLORS from 'common/constants/colors';
+import { messages } from './messages';
+
+const EXECUTIONS = 'executions';
+
+const range = (start = 0, end = 0, step = 1) => {
```
[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/components/widgets/charts/common/utils.js]: Done

File: app/localization/translated/be.json [hunk 9]
```diff
@@ -280,20 +281,13 @@
...
   "ExternalUserInvitationModal.header": "Запрасіць карыстальніка",
   "ExternalUserInvitationModal.link": "Спасылка на запрашэнне",
+  "FailedCasesTrendChart.failedCases": "няўдалыя сцэнарыі",
   "Filter.add": "Дадаць фільтр",
+  "Filter.description": "Апісанне",
   "Filter.descriptionPlaceholder": "Увядзіце апісанне фільтра",
   "Filter.edit": "Рэдагаваць фільтр",
   "Filter.name": "Назва",
   "Filter.namePlaceholder": "Увядзіце назву фільтра",
-  "FilterNameById.statistics$defects$automation_bug": "Хіба аўтаматызацыі",
```
Comment (DiscoElevator) [inline app/localization/translated/be.json]: Does it need to be removed?

File: app/src/components/widgets/charts/launchStatisticsChart/launchStatisticsChart.jsx [hunk 10]
```diff
@@ -345,6 +299,83 @@ export class LaunchStatisticsChart extends Component {
     };
   };
 
+  setupConfigData = (data, isTimeLine) => {
+    const {
+      defectTypes,
+      widget: {
+        content_parameters: { content_fields, widgetOptions },
```
Comment (DiscoElevator) [inline app/src/components/widgets/charts/launchStatisticsChart/launchStatisticsChart.jsx]: camelCase

File: app/localization/translated/be.json (hunk 9 above)
Comment (AmsterGet) [reply app/localization/translated/be.json]: Yes, we don't use it in our code now

File: app/src/components/widgets/charts/launchStatisticsChart/launchStatisticsChart.jsx (hunk 10 above)
Comment (AmsterGet) [reply app/src/components/widgets/charts/launchStatisticsChart/launchStatisticsChart.jsx]: Done

**Pull request merged**